### PR TITLE
[DO NOT MERGE] Adds 'How to Contribute Documentation' doc

### PR DESCRIPTION
**Pull request by kberryUSGS**
Adds a "How to Contribute Documentation" document for discussion and feedback.

Please do not merge. This PR exists only for feedback on the documentation. The final outcome from any feedback gathered will be an update to the following wiki page:
https://github.com/USGS-Astrogeology/ISIS3/wiki/Contributing-Application-Documentation
and any of the other wiki pages it links to.

### CODE REVIEW

File: ContributeDocumentation.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Contributing Application Documentation Changes
+
+This document describes how to contribute changes to the ISIS Application Documentation, which is available online for the most recent ISIS release at: https://isis.astrogeology.usgs.gov/Application/index.html
+
+# Getting Started
+Documentation for ISIS applications is located within the ISIS source code and written in XML. The XML is processed to create the documentation that can be viewed on the ISIS website at: https://isis.astrogeology.usgs.gov/Application/index.html
+
+For many documentation contributions, very little understanding of XML is required. Changes in existing wording or additional information can be added where appropriate in the current documentation without needing to change any of the XML.
```
Comment (scsides) [inline ContributeDocumentation.md]: Unless I'm misunderstanding, the second sentence 
isn't true.

File: ContributeDocumentation.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+# Contributing Application Documentation Changes
+
+This document describes how to contribute changes to the ISIS Application Documentation, which is available online for the most recent ISIS release at: https://isis.astrogeology.usgs.gov/Application/index.html
+
+# Getting Started
+Documentation for ISIS applications is located within the ISIS source code and written in XML. The XML is processed to create the documentation that can be viewed on the ISIS website at: https://isis.astrogeology.usgs.gov/Application/index.html
```
Comment (scsides) [inline ContributeDocumentation.md]: This makes it sound like the docs are embedded in the cpp files. Suggest reword to say it is in the directory for each app with the same name except .xml

File: ContributeDocumentation.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+Since the ISIS documentation is located within the source code, the first step is to create a local copy of the ISIS source code to work with. To do the necessary initial setup, first complete the steps in [Getting Started with Github](https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#getting-started-with-github), then move on to [Anaconda and ISIS3 Dependencies](https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#getting-started-with-github). 
+ 
+# Making Your Documentation Change
+
+Documentation for each ISIS application is located in the same directory as the source code for the application itself and is named `name_of_application.xml` For example, to update the documentation for voy2isis, go to `ISIS3/isis/src/voyager/apps/voy2isis` in your clone and begin editing `voy2isis.xml`. To make your documentation change, open the xml file in the text editor of your choice, and edit the text to make your update. Most documentation changes will be made in the `<description>` section of the XML file, which creates the "Description" section on the built html documentation.
```
Comment (scsides) [inline ContributeDocumentation.md]: Maybe reword as suggested above and/or move some of the up or reorganize

File: ContributeDocumentation.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+# How to Preview Your Change
+
+In order to preview your documentation change, it is necessary to build the documentation and view it in a web browser.
```
Comment (scsides) [inline ContributeDocumentation.md]: I guess we lost the make html ability. That is too bad

File: ContributeDocumentation.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+`conda activate <environment-name>`
+
+* Navigate to the top-level directory of your ISIS3 clone (i.e `cd ISIS3`)
+
+* Create a `build` and an `install` directory at this level:
+  * `mkdir build install`
+
+* Navigate to the build directory: `cd build` 
+* Run the following command to configure your build:
```
Comment (scsides) [inline ContributeDocumentation.md]: May want to refer to the main docs for this. If something changes, and we don't look everywhere for how to config we'll miss this one and it will be wrong. Suggest never duplicating docs

File: ContributeDocumentation.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+
+## Previewing the Built Documentation
+To view your documentation, open the `index.html` file immediately under the docs directory (`$ISISROOT/docs/index.html`) using a web browser and use the web browser to navigate to the modified file. It is possible to do this by:
+
+* running either of the following in a terminal: `google-chrome index.html` or `firefox index.html`
+* copy-and-pasting the full file path to `index.html` (for example: `/home/username/dev/ISIS3/isis-build/docs/index.html`) into the address bar of a web browser.
+
+
+If you see changes that need to be made to the documentation at this point, you can make the changes in your documentation XML files, re-run your documentation build command, and then just refresh the web-page in your browser to view your updated changes.
```
Comment (scsides) [inline ContributeDocumentation.md]: With only changes to one app.xml, this should be quick, but is it possible to specify the exact target "ninja spiceinit.html"

File: ContributeDocumentation.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+* running either of the following in a terminal: `google-chrome index.html` or `firefox index.html`
+* copy-and-pasting the full file path to `index.html` (for example: `/home/username/dev/ISIS3/isis-build/docs/index.html`) into the address bar of a web browser.
+
+
+If you see changes that need to be made to the documentation at this point, you can make the changes in your documentation XML files, re-run your documentation build command, and then just refresh the web-page in your browser to view your updated changes. 
+
+# Adding Your Documentation Changes to ISIS
+To contribute your documentation changes to ISIS, please put in a pull request containing your documentation changes, following the instructions at [How to Put in a PR](https://github.com/USGS-Astrogeology/ISIS3/wiki/How-to-Start-Contributing#create-a-pull-request). Be sure to read the [Contributing](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document, and to select the "Documentation change" under "Types of change" in the PR template which will appear when you put in your pull request. Your pull request will be reviewed by the software team at the Astrogeology Science Center.
```
Comment (scsides) [inline ContributeDocumentation.md]: With to goals of the TC, we should use a general reference to all devs instead of Astro

File: ContributeDocumentation.md (hunk 4 above)
Comment (kberryUSGS) [reply ContributeDocumentation.md]: Yeah :(

File: ContributeDocumentation.md (hunk 1 above)
Comment (kberryUSGS) [reply ContributeDocumentation.md]: Thanks for pointing this out! My intention was to communicate that you could edit the wording of the documentation without editing/modifying the xml tags themselves because this originally was aimed at a larger (and potentially less-technical) audience, but you're right and this needs to be fixed. Not sure how to update the wording yet, or maybe I'll just take this out.

File: ContributeDocumentation.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+# Getting Started
+Documentation for ISIS applications is located within the ISIS source code and written in XML. The XML is processed to create the documentation that can be viewed on the ISIS website at: https://isis.astrogeology.usgs.gov/Application/index.html
+
+For many documentation contributions, very little understanding of XML is required. Changes in existing wording or additional information can be added where appropriate in the current documentation without needing to change any of the XML. 
+
+Since the ISIS documentation is located within the source code, the first step is to create a local copy of the ISIS source code to work with. To do the necessary initial setup, first complete the steps in [Getting Started with Github](https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#getting-started-with-github), then move on to [Anaconda and ISIS3 Dependencies](https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#getting-started-with-github).
```
Comment (AndrewAnnex) [inline ContributeDocumentation.md]: is it actually necessary to follow the 2nd set of instructions, naively my expectation for contributing a doc change would be to clone the ISIS3 repo then edit the XML file/source code then push the code as a branch then PR. Is this document also intending to be a guide for developers contributing new applications to ISIS3?

**Comment by kberryUSGS** [comment]
Sorry @rfergason didn't mean to re-request a review from you, since you already approved it, but accidentally hit the wrong button.

**Comment by kberryUSGS** [comment]
I'm going to leave this open for a little longer in case anyone else has any comments, now that I've updated it for @scsides's feedback.

**Comment by kberryUSGS** [comment]
@jessemapel Definitely agreed that we still need documentation on how to contribute code changes :+1:

**Comment by jessemapel** [comment]
At the last meeting it was agreed this was good to go. @kberryUSGS can you open a PR on ISIS with this?
